### ISSUE TITLE
Create Ultimate Mode

### ISSUE THREAD

**Issue by Torvec** [Torvec/tic-tac-toe-react#5]
Should contain the following:
- [x] Randomly selects who starts (X or O)
- [x] Display 9 3 x 3 grids
- [x] Starting from the center 3x3 grid, depending on what cell is selected you then go to the grid that is in the same location relative to the center 3x3 grid
```
[ 1 ] [ 2 ] [ 3 ]
[ 4 ] [ 5 ] [ 6 ]
[ 7 ] [ 8 ] [ 9 ]
```
  - For example, the player selects position 3 on the center grid, the next turn needs to choose a cell in the game grid on the top right corner. Each move follows the same rule unless a grid has already been won then the current player can choose any available grid.
- [x] If a corresponding grid is unplayable (i.e its won or draw), the player can play on any playable grid
- [x] After the player makes their move it should go back to only one active grid at a time unless the next grid is unplayable
- [x] Each cell in a grid starts as empty, clicking it changes it to the current player's symbol
- [x] Can only click on an empty cell, clicking an occupied cell does nothing (including not changing the turn)
- [x] Current player changes when a cell is filled
- [x] Win Conditions:
  - 3 in a row/column/diagonal per game grid gives that player the win for that game grid
  - 3 game grids in a row/column/diagonal gives that player the overall win
- [x] Draw Condition: 
  - All grids are in an unplayable state (X won, O won, or Draw)
- [x] Win or Draw Condition stops the game
- [x] Win Message after win condition met
- [x] Draw Message after draw condition met

**Comment by Torvec** [comment]
All the logic for changing which grid is active, inactive, notPlayable, etc works except for one specfic scenario where a winning move is made in a cell that corresponds to the same grid (i.e cell 6 in grid 6, cell 4 in grid 4, etc.)

**Comment by Torvec** [comment]
Another issue found: Game recognizes when the board is won but you can still keep playing, needs to stop the game entirely at that point. Probably the same for draw game too

**Comment by Torvec** [comment]
And another issue just showed up where if all grids become unplayable but there is a winning combo made on the last move it shows up as a win and a draw

**Comment by Torvec** [comment]
Found another scenario where things don't quite work right, if there is only one playable grid left it will make that grid inactive when you make a move and then you can't complete the game.

**Comment by Torvec** [comment]
> And another issue just showed up where if all grids become unplayable but there is a winning combo made on the last move it shows up as a win and a draw

I wasn't doing if else if for the messages, just if and if for the win and draw messages in render()

**Comment by Torvec** [comment]
> Found another scenario where things don't quite work right, if there is only one playable grid left it will make that grid inactive when you make a move and then you can't complete the game.

Should be working now, just had to make it so the previous grid was not inactivated if the next grid was not playable

**Comment by Torvec** [comment]
> All the logic for changing which grid is active, inactive, notPlayable, etc works except for one specfic scenario where a winning move is made in a cell that corresponds to the same grid (i.e cell 6 in grid 6, cell 4 in grid 4, etc.)

Ended up being some complicated if else, else if statements needed to be made to cover every possible scenario. Spent way too much time trying to figure this part out...

**Comment by Torvec** [comment]
> Another issue found: Game recognizes when the board is won but you can still keep playing, needs to stop the game entirely at that point. Probably the same for draw game too

Handled it in the handleBoardStateChange method

**Comment by Torvec** [comment]
It is more or less done, just have to integrate it with the rest of the UI once that part is finished
